packages/plugin-types/src/deep_partial.ts

```diff
@@ -5,3 +5,17 @@
 export type DeepPartial<T> = {
   [P in keyof T]?: DeepPartial<T[P]>;
 };
+
+/**
+ * Makes all the properties of the entire nested object partial for Amplify generated configs
+ * instead of just the top level properties. Other properties are not changed.
+ */
+export type DeepPartialAmplifyGeneratedConfigs<T> = {
+  [P in keyof T]?: P extends 'analytics' | 'geo' | 'notifications'
+    ? T[P]
+    : (
+        T[P] extends object
+          ? DeepPartialAmplifyGeneratedConfigs<T[P]>
+          : Partial<T[P]>
+      );
+};
```

Comment:
I'd flip the logic, i.e. make `auth`, `storage` and `data` DeepPartial and everything else leave it as it is. The reason is that in future when we extend to other categories and config, it's better to start more restrictive, i.e. required than optional. (Going from required to optional is not a breaking change but other way around is)

Also, can we pass this `'auth' | 'storage' | 'data'` as a parameter to this type?

Reply:
Sure I can flip the logic to be more restrictive. 

As for passing them as a parameter, we would have to maintain this list everywhere the type is being used (which is 10 places currently) and some of these places can be missed when updating this list. This is why I chose to remove the parameter so we can maintain the list in one place.

Reply:
This list can come from `backend-output-schemas` or even `client-config` package. It's not a blocker and we can change it later if needed.